aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -57,6 +57,86 @@ TORCH_META_FUNC(mm)(const Tensor & self, const Tensor & mat2) {
   "The input tensor must be a matrix with size ", self.sizes()[0], "x", mat2.sizes()[1], ", but got a ", result.dim(),
   "-D tensor with size ", result.sizes()[0], "x", result.sizes()[1]);
 }
+
+template <typename Meta>
+void common_checks_baddbmm_bmm(Meta& meta, const Tensor& batch1, const Tensor& batch2, const Scalar& beta, const Scalar& alpha, bool is_bmm, const c10::optional<Tensor>& self_or_result=nullopt) {
+  CheckedFrom c = (is_bmm ? "bmm" : "baddbmm");
+
+  auto checkOnCPU = [](const Tensor& t, CheckedFrom c) {
+    TORCH_CHECK(
+      !t.is_cuda(),
+      "Expect tensor to have CPU backend, but got tensor with ",
+      toString(t.options().backend()),
+      " Backend (while checking arguments for ",
+      c);
+  };
+
+  if (self_or_result.has_value()) checkOnCPU(self_or_result.value(), c);
+  checkOnCPU(batch1, c);
+  checkOnCPU(batch2, c);
```

Comment:
I think checking whether tensors are in the same device is already taken care of because of how dispatch works (given that this function is only called for CPU, CUDA, and META backends).

Reply:
I think these checks were useful since the implementations are different, so just to make sure no one in future calls this impl for CUDA. I've removed these checks from the meta function, but have kept them for impl, for consistency and correctness in the future.